src/zenml/metadata_stores/mysql_metadata_store.py

```diff
@@ -37,11 +42,93 @@ def get_tfx_metadata_config(
+        else:
+            if secret and secret.user:
+                config.user = self.user
+            else:
+                raise RuntimeError(
+                    f"Your metadata store does not have a username. Please "
+                    f"provide it either by defining it upon registration or "
+                    f"through a MySQL secret."
+                )
+
+        # Set the password
+        if self.password:
+            if secret and secret.password:
+                raise RuntimeError(
+                    f"Both the metadata store {self.name} and the secret "
+                    f"{self.secret} within your secrets manager define "
+                    f"a password. Please make sure that you only use one."
+                )
+            else:
+                config.password = self.password
+        else:
+            if secret and secret.user:
+                config.password = self.password
+
+        # Set the SSL configuration if there is one
+        if secret:
+            ssl_options = {}
+            # Handle the files
+            for key in ["ssl_key", "ssl_ca", "ssl_cert"]:
+                content = getattr(secret, key)
+                filepath = os.path.join(os.getcwd(), f"{key}.pem")
+                ssl_options[key.lstrip("ssl_")] = filepath
+                with open(filepath, "w") as f:
+                    f.write(content)
+
+            # Handle additional params
+            ssl_options["verify_server_cert"] = secret.ssl_verify_server_cert
+
+            ssl_options = MySQLDatabaseConfig.SSLOptions(**ssl_options)
+            config.ssl_options.CopyFrom(ssl_options)
+
+        return metadata_store_pb2.ConnectionConfig(mysql=config)
+
+    def _get_mysql_secret(self) -> Any:
+        """Method which returns a MySQL secret from the secrets manager."""
+        if self.secret:
+            active_stack = Repository().active_stack
+            secret_manager = active_stack.secrets_manager
+            if secret_manager is None:
+                raise RuntimeError(
+                    f"The metadata store `{self.name}` that you are using "
+                    f"requires a secret. However, your stack "
+                    f"`{active_stack.name}` does not have a secrets manager."
+                )
+            try:
+                secret = secret_manager.get_secret(self.secret)
+
+                if isinstance(secret, MYSQLSecretSchema):
+                    raise RuntimeError(
+                        f"If you are using a secret with a MySQL Metadata "
+                        f"Store, please make sure to use the schema: "
+                        f"{MYSQLSecretSchema.TYPE}"
+                    )
+            except KeyError:
+                raise RuntimeError(
+                    f"The secret `{self.secret}` used for your MySQL metadata "
+                    f"store `{self.name}` does not exist in your secrets "
+                    f"manager `{secret_manager.name}`."
+                )
```

Comment:
I think this is also missing a `return secret` somewhere, currently it always returns `None`

Reply:
Good catch! Done!